pyuvdata/tests/test_uvbeam.py

```diff
@@ -403,7 +414,15 @@ def test_interpolation():
 
     interp_data_array, interp_basis_vector = power_beam.interp(az_array=az_interp_vals,
                                                                za_array=za_interp_vals,
-                                                               freq_array=freq_interp_vals)
+                                                               freq_array=freq_interp_vals)    
+
+    # test if fed original frequencies that not interp_bool is triggered
+    # by using only one freq chan in object, which should trigger a ValueError if interp_bool is True
+    _pb = power_beam.select(frequencies=power_beam.freq_array[0, :1], inplace=False)
+    try:
+        interp_data_array, interp_bandp = _pb._interp_freq(_pb.freq_array[0], kind='linear')
+    except:
+        raise AssertionError("UVBeam._interp_freq didn't return array slice as expected...")
```

Comment:
The point is, if with the inputs given, `UVBeam._interp_freq` should error if it interpolate the beam, rather than slices it so the test is actually supposed to test that it *doesn't* raise an error in this case, and this is the clearest way to do that in my mind. I could just make the function call and not check any of the outputs, but I've gotten pushback on that from other people before, so the `try/except` with an error message is in my mind is more of a clear way to do it. Lmk if you have any suggestions.

Reply:
I guess I would just do the call and then assert that the returned array matched the data_array on the object. But I'm ok with this now that you've added the comments to explain what you're doing.